### DISCUSSION TITLE
Motorized Faders and problems with Cubase

### DISCUSSION THREAD

**Discussion by Vahen1981** [tttapa/Control-Surface-Motor-Fader#2] [General]
Hi Pieter!
Me again, this time without dubts at least haha, but I want to share something.
I was trying your code for motorized faders, and first I have to thank you for this, it works very good. I was testing it with cubase and ableton. For Ableton was working perfect, but for Cubase it has an issue that I could fix easily, but after hours of thinking haha.

The problem was that apparently cubase doesn't loops back the pitchbend messages when you use the real fader. Everything is fine when you are using the virtual fader, it sends messages to the controller and then the motorized fader moves perfectly... but when you move the real fader this returns to the last position where the virtual fader was moved, and this is a real problem for control the DAW.

Well the thing is , that I easily fixed with midi pipes adding this lines to the void setup() :

```
Control_Surface << pipe1 << midiusb;
Control_Surface >> pipe2 >> midiusb;
Control_Surface >> pipe3 >> Control_Surface;
midiusb.begin();
```

I'm not so sure what I did, but at least my intention was to loops back the pitchbend midi message that the real fader it's sending, since the DAW is not sending back this message.

What do you think about this method? Maybe should be a better option, but at least for me works good untill now. Thank you Pieter!

**Comment by tttapa** [comment]
> apparently cubase doesn't loops back the pitchbend messages when you use the real fader

Could this be because of the automation mode? I'm not familiar with Cubase, but this makes a difference in Reaper.  
https://steinberg.help/cubase_pro_artist/v9/en/cubase_nuendo/topics/automation/automation_writeread_automation_c.html  
Is automation write (or latch?) enabled?

> What do you think about this method?

I think it's a bit wasteful to send everything through Control_Surface again, that adds quite a bit of unnecessary overhead. You might want to change the setpoint in the sketch directly.

**Comment by Vahen1981** [comment]
No, is not the automation, read or write, enable or disable the fader has the same behavior.
I thought the same... maybe after some minutes the system colapse... I tried the setpoint before without succes, but now you say that is an option I'll keep trying. Thanks for your advice!
